spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -179,11 +179,17 @@ public void setClientIdSuffix(String clientIdSuffix) {
 	 * either explicitly or by Kafka; may be null if not assigned yet.
 	 */
 	public Collection<TopicPartition> getAssignedPartitions() {
-		if (this.listenerConsumer.definedPartitions != null) {
-			return Collections.unmodifiableCollection(this.listenerConsumer.definedPartitions.keySet());
-		}
-		else if (this.listenerConsumer.assignedPartitions != null) {
-			return Collections.unmodifiableCollection(this.listenerConsumer.assignedPartitions);
+		ListenerConsumer listenerConsumer = this.listenerConsumer;
+		if (listenerConsumer != null) {
+			if (listenerConsumer.definedPartitions != null) {
+				return Collections.unmodifiableCollection(listenerConsumer.definedPartitions.keySet());
+			}
+			else if (listenerConsumer.assignedPartitions != null) {
+				return Collections.unmodifiableCollection(listenerConsumer.assignedPartitions);
+			}
+			else {
+				return null;
```

Comment:
I would suggest to fix `toString()` as well.
Right now it might finish with the `", topicPartitions=null"`, which might not be so useful.

Your thoughts?

Thanks